apps/common/behaviours/pdf-upload.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const UploadModel = require('../../common/models/file-upload');
+const PDFModel = require('../../common/models/pdf');
+
+module.exports = superclass => class PDFUpload extends superclass {
+
+  process(req, res, next) {
+    this._render(req, res)
+      .then(html => this._create(html))
+      .then(pdfBuffer => this._upload({
+        name: 'application_form.pdf',
+        data: pdfBuffer,
+        mimetype: 'application/pdf'
+      }))
+      .then(result => {
+        req.form.values['pdf-upload'] = result.url;
+      })
+      .then(() => {
+        super.process(req, res, next);
+      }, next)
+      .catch((err) => {
+        next(err);
+      });
+  }
+
+  _render(req, res) {
```

Comment:
Not super-thrilled about using `_render` as a method name because of `render` already being a distinct part of the request pipeline, and the leading underscore being an established hof pattern for private versions of extensible methods. It leads to the incorrect impression that this is related to the existing `render`phase of the request pipeline.

It *probably* doesn't matter *too much* for firearms, but if it's easy to update I'd prefer `renderPDF` (and possibly equivalent `createPDF` and `uploadPDF` methods).